lib/gc/Transforms/OneDNNGraphToLinalg.cpp

```diff
@@ -270,23 +493,56 @@ struct MatMulOpLowering : public OpRewritePattern<MatMulOp> {
 
 struct ConvertOneDNNGraphToLinalg
     : public impl::ConvertOneDNNGraphToLinalgBase<ConvertOneDNNGraphToLinalg> {
-
   void runOnOperation() final {
     auto *ctx = &getContext();
-    // add lowering target
+    // ==========================================
+    // perform pre conversion
+    // ==========================================
+    RewritePatternSet patternsPre(ctx);
+    patternsPre.add<
+        // clang-format off
+        MatMulOpBatchFlatten
+        // clang-format on
+        >(ctx);
+    if (failed(applyPatternsAndFoldGreedily(getOperation(),
+                                            std::move(patternsPre)))) {
+      signalPassFailure();
+    }
+    // ==========================================
+    // perform conversion
+    // ==========================================
     ConversionTarget target(getContext());
     target.addIllegalDialect<onednn_graph::OneDNNGraphDialect>();
-    target.addLegalDialect<BuiltinDialect, arith::ArithDialect,
-                           linalg::LinalgDialect, func::FuncDialect,
-                           tensor::TensorDialect>();
-    // set pattern
+    target.addLegalDialect<
+        // clang-format off
+        BuiltinDialect, 
+        scf::SCFDialect,
```

Comment:
What are the cases we produce scf during the lowering?

Reply:
Used to test something, now removed